src/bwamem_pair.cpp

```
@@ -48,6 +48,121 @@ Authors: Vasimuddin Md <vasimuddin.md@intel.com>; Sanchit Misra <sanchit.misra@i
+        if (collapse == 1 && b == 1) return 3;   /* C -> T */
+        if (collapse == 2 && b == 2) return 0;   /* G -> A */
+        return b;
+    };
+
+    /* oriented (RC for is_rev), collapsed query bases */
+    static thread_local std::vector<uint8_t> q;
+    if ((int)q.size() < l_ms) q.resize(l_ms);
+    if (is_rev) for (int i = 0; i < l_ms; ++i) { uint8_t c = ms[l_ms-1-i]; q[i] = col((c < 4)? 3-c : 4); }
+    else        for (int i = 0; i < l_ms; ++i) q[i] = col(ms[i]);
+
+    /* index the read's k-mers in a fixed open-addressing hash (O(read) build,
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)